lib/rule-tester/rule-tester.js

```diff
@@ -640,6 +641,165 @@ class RuleTester {
             });
         });
     }
+
+    /**
+     * Generate a snapshot test for the given code and rule
+     * @param {Function} rule The rule to test.
+     * @param {Object} testCase The collection of tests to run.
+     * @returns {void}
+     */
+    generateSnapshot(rule, testCase) {
+        const testerConfig = this.testerConfig;
+        const linter = this.linter;
+        const ruleName = "snapshot";
+
+        linter.defineRule(ruleName, Object.assign({}, rule, {
+
+            // Create a wrapper rule that freezes the `context` properties.
+            create(context) {
+                freezeDeeply(context.options);
+                freezeDeeply(context.settings);
+                freezeDeeply(context.parserOptions);
+
+                return (typeof rule === "function" ? rule : rule.create)(context);
+            }
+        }));
+
+        linter.defineRules(this.rules);
+
+        /**
+         * Run the rule for the given item
+         * @param {string|Object} item Item to run the rule against
+         * @returns {Object} Eslint run result
+         * @private
+         */
+        function runRuleForItem(item) {
```

Comment:
this is a copy & paste.
I think it would be better ti refactor it later in a separate PR?
So that you can focus on review the new feature.